File: benchmarks/src/lua/trie_dict.lua
local W=20000;local seed=12345;local function ri(m)seed=(seed*16807)%2147483647;return seed%m end
local nodes=1;local children={};for i=1,26 do children[i]=0 end;local terminal={0}
local function nc()nodes=nodes+1;for i=1,26 do children[#children+1]=0 end;terminal[#terminal+1]=0;return nodes-1 end
local start=os.clock()
for _=1,W do local l=3+ri(6);local cur=0
 for _=1,l do local c=ri(26);local idx=cur*26+c+1
  if children[idx]==0 then children[idx]=nc()end;cur=children[idx]end
 terminal[cur+1]=1 end
local hits=0;seed=12345
for _=1,W do local l=3+ri(6);local cur=0
 for _=1,l do local c=ri(26);local idx=cur*26+c+1
  if children[idx]==0 then cur=0;break end;cur=children[idx]end
 if cur~=0 and terminal[cur+1]==1 then hits=hits+1 end end
seed=54321
for _=1,W do local l=3+ri(6);local cur=0
 for _=1,l do local c=ri(26);local idx=cur*26+c+1
  if children[idx]==0 then cur=0;break end;cur=children[idx]end
 if cur~=0 and terminal[cur+1]==1 then hits=hits+1 end end

local finish=os.clock()
print("Result: "..nodes.."_"..hits)
print(string.format("Time: %.0fms",(finish-start)*1000))

File: benchmarks/src/python/trie_dict.py
import time
W=20000;seed=12345
def ri(m):
 global seed;seed=(seed*16807)%2147483647;return seed%m
nodes=1;children=[0]*26;terminal=[0]
def nc():
 global nodes,children,terminal
 for _ in range(26):children.append(0)
 terminal.append(0);nodes+=1;return nodes-1
start=time.time()*1000
for _ in range(W):
 l=3+ri(6);cur=0
 for _ in range(l):
  c=ri(26);idx=cur*26+c
  if children[idx]==0:children[idx]=nc()
  cur=children[idx]
 terminal[cur]=1
hits=0;seed=12345
for _ in range(W):
 l=3+ri(6);cur=0
 for _ in range(l):
  c=ri(26);idx=cur*26+c
  if children[idx]==0:cur=0;break
  cur=children[idx]
 if cur!=0 and terminal[cur]==1:hits+=1
seed=54321
for _ in range(W):
 l=3+ri(6);cur=0
 for _ in range(l):
  c=ri(26);idx=cur*26+c
  if children[idx]==0:cur=0;break
  cur=children[idx]
 if cur!=0 and terminal[cur]==1:hits+=1
end=time.time()*1000
print(f"Result: {nodes}_{hits}")
print(f"Time: {int(end-start)}ms")